lib/idempo.rb

```diff
@@ -57,6 +58,12 @@ def call(env)
       status, headers, body = @app.call(env)
 
       expires_in_seconds = (headers.delete("X-Idempo-Persist-For-Seconds") || @persist_for_seconds).to_i
+
+      # In some cases `body` could respond to to_ary. In this case, we don't need to call .close on body afterwards.
+      #
+      # @see https://github.com/rack/rack/blob/main/SPEC.rdoc#the-body-
+      body = body.try(:to_ary) if rack_v3? && body.respond_to?(:to_ary)
```

Comment:
nit: since you already do `respond_to?` you don't need `try() you can call `to_ary` directly